src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -25,29 +25,32 @@ def event_activate(self):
             f'{self.channel_prefix}/device_info': self._device_info,
         })
 
-    def _list_bids(self, _):
+    def _list_bids(self, payload):
         list_bids_response_channel = f'{self.channel_prefix}/response/list_bids'
         if not check_for_connected_and_reply(self.redis, list_bids_response_channel,
                                              self.connected):
             return
+        arguments = json.loads(payload["data"])
         self.pending_requests.append(
-            IncomingRequest("list_bids", None, list_bids_response_channel))
+            IncomingRequest("list_bids", arguments, list_bids_response_channel))
 
-    def _list_bids_impl(self, _, response_channel):
+    def _list_bids_impl(self, arguments, response_channel):
         try:
             filtered_bids = [{"id": v.id, "price": v.price, "energy": v.energy}
                              for _, v in self.market.get_bids().items()
                              if v.buyer == self.device.name]
             self.redis.publish_json(
                 response_channel,
-                {"command": "list_bids", "status": "ready", "bid_list": filtered_bids})
+                {"command": "list_bids", "status": "ready", "bid_list": filtered_bids,
+                 "transaction_id": arguments["transaction_id"]})
```

Comment:
If a message with no transaction id is passed, I think the redis thread will crash, In this case a KeyError will be raised, which will get caught by the `except` block. But, since the except block also uses `arguments["transaction_id"]` , it will also raise an exception which will not get caught, and the redis thread that receives the incoming messages will terminate.